.github/workflows/dependabot.yaml

```diff
@@ -0,0 +1,91 @@
+name: dependabot
+
+on:
+  pull_request:
+    types:
+      - opened
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+# Only run one instance per PR to ensure in-order execution.
+concurrency: pr-${{ github.ref }}
+
+jobs:
+  # Dependabot is annoying, but this makes it a bit less so.
+  dependabot-automerge:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request' && github.event.pull_request.user.login == 'dependabot[bot]' && github.repository == 'coder/coder'
+    permissions:
+      pull-requests: write
+      contents: write
```

Comment:
Do we need this given that `pull-requests: write` is granted?

Reply:
Yes. It's needed to approve and merge PRs

Reply:
Got a reference for that?

Reply:
https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions#enabling-automerge-on-a-pull-request

I have also tested by removing these permissions and adding them as required. The merge action was failing without them.

Reply:
Added an additional check for `user_id`
Got it from: https://api.github.com/users/dependabot[bot].

Reply:
Sweet thanks 👍